### PR TITLE
Perf: don't run the regex replace unless the selectorText contains a colon

### PR DESCRIPTION
**Pull request by eoghanmurray**
(rules generally contain colons)

This is an imagined performance improvement on #1253 so should be measured!

### CODE REVIEW

File: packages/rrweb-snapshot/src/utils.ts [hunk 1]
```diff
@@ -75,25 +75,27 @@ export function getCssRuleString(rule: CSSRule): string {
     } catch {
       // ignore
     }
+  } else if (isCSSStyleRule(rule) && rule.selectorText.includes(':')) {
+    // Safari does not escape selectors with : properly
```
Comment (eoghanmurray) [inline packages/rrweb-snapshot/src/utils.ts:79]:
```suggestion
    // Safari does not escape selectors with : properly
    // see https://bugs.webkit.org/show_bug.cgi?id=184604    
```

**Comment by mydea** [comment]
Looks good to me - I also had a hard time testing this locally, but this should be good I think (don't have any setup right now to verify this, but the existing test _should_ cover this hopefully?)

**Comment by eoghanmurray** [comment]
Checks are failing as I authored it with #1279 in place, so that should be considered a prerequisite of this

**Review by Juice10 — APPROVED** [review]

**Comment by Juice10** [comment]
@eoghanmurray could you rebase this PR, I just merged in #1279

**Pull request merged**